File: src/java/magmac/app/compile/rule/split/DividingSplitter.java
package magmac.app.compile.rule.split;

import magmac.api.Option;
import magmac.api.Tuple2;
import magmac.api.collect.list.List;
import magmac.api.iter.collect.ListCollector;
import magmac.app.compile.rule.Splitter;
import magmac.app.compile.rule.divide.Divider;

public final class DividingSplitter implements Splitter {
    private final Divider divider;
    private final Selector selector;

    private DividingSplitter(Divider divider, Selector selector) {
        this.divider = divider;
        this.selector = selector;
    }

    public static Splitter Last(Divider divider) {
        return new DividingSplitter(divider, new LastSelector());
    }

    public static Splitter First(Divider divider) {
        return new DividingSplitter(divider, new FirstSelector());
    }

    @Override
    public Option<Tuple2<String, String>> split(String input) {
        List<String> list = this.divider.divide(input).collect(new ListCollector<>());
        return this.selector.select(list);
    }

    @Override
    public String createMessage() {
        return "Insufficient segments present";
    }

    @Override
    public String merge(String leftString, String rightString) {
        return leftString + " " + rightString;
    }
}

File: src/web/magmac/app/compile/rule/split/DividingSplitter.ts
import { Option } from "../../../../../magmac/api/Option";
import { Tuple2 } from "../../../../../magmac/api/Tuple2";
import { List } from "../../../../../magmac/api/collect/list/List";
import { ListCollector } from "../../../../../magmac/api/iter/collect/ListCollector";
import { Splitter } from "../../../../../magmac/app/compile/rule/Splitter";
import { Divider } from "../../../../../magmac/app/compile/rule/divide/Divider";
export class DividingSplitter {
	DividingSplitter(divider : Divider, selector : Selector) : private {break;break;;}
	Last(divider : Divider, delimiter : String) : Splitter {return new DividingSplitter( 0, new LastSelector( 0));;}
	First(divider : Divider, delimiter : String) : Splitter {return new DividingSplitter( 0, new FirstSelector( 0));;}
	split(input : String) : Option<Tuple2<String, String>> {break;return 0.selector.select( 0);;}
	createMessage() : String {return "Insufficient segments present";;}
	merge(leftString : String, rightString : String) : String {return 0;;}
}
